hw/ip/usbdev/data/usbdev_testplan.hjson

```diff
@@ -18,12 +19,824 @@
+      tests: ["usbdev_device_address"]
+    }
+    {
+      name: link_in_err
+      desc: '''
+            Verify that if a packet to an IN endpoint started to be received but was then dropped due to an error
+            then link_in_err will raise.
+
+            - send an IN token to the device for initiating an IN transaction.
+            - The device should respond with IN data packets.
+            - In the handshake send an invalid CRC or a different PID then expected.
+            - Verify that the link_in_err should go high following the handshake packet sent by the host.
+            '''
+      stage: V1
+      tests: ["usbdev_link_in_err"]
+    }
+    {
+      name: rx_crc_err
+      desc: '''
+            Verify that if a CRC error occurred then rx_crc_err will raise.
+
+            - send a packet(token/handshake/data) to the device while intentionally corrupting any of the CRC bits.
+            - Verify that the rx_crc_err should go high following the reception of erroneous packet.
+            '''
+      stage: V1
+      tests: ["usbdev_rx_crc_err"]
+    }
+    {
+      name: rx_pid_err
+      desc: '''
+            Verify that if an invalid packed identifier (PID) was received then rx_pid_err will raise.
+
+            - send an invalid PID to the device with the higher bits not complementing the lower bits.
+            - Verify that the interrupt pin goes high indicating PID error.
+            '''
+      stage: V1
+      tests: ["usbdev_rx_pid_err"]
+    }
+    {
+      name: rx_bitstuff_err
+      desc: '''
+            Verify that if an invalid bit stuffing was received then rx_bitstuffing_err will raise.
+
+            - generate a packet with 7 consecutive ones (bypassing bitstufffing) on data line.
+            - Verify that the interrupt pin goes high indicating bitstuff error.
+            '''
+      stage: V1
+      tests: ["usbdev_rx_bitstuff_err"]
+    }
+    {
+      name: link_out_err
+      desc: '''
+            Verify that if a packet to an OUT endpoint started to be received but was then dropped due to an error
+            then link_out_err will raise.
+
+            - send an out packet to the device with invalid crc, token, avbuffer_empty, rx_fifo full conditions.
+            - Verify that the interrupt pin goes high indicating an error in the link in OUT transaction.
+            '''
+      stage: V1
+      tests: ["usbdev_link_out_err"]
+    }
+    {
+      name: invalid_data1_data0_toggle_test
+      desc: '''
+            Verify the functionality of invalid toggling of Data1 and Data0 bits.
+
+            - send an out data packet to the device in an ongoing transaction without toggling data0 and data1
+              PIDs in consecutive data packets.
+            - Verify that the interrupt pin goes high indicating an error in the link while OUT transaction.
+            '''
+      stage: V1
+      tests: ["usbdev_invalid_data1_data0_toggle_test"]
+    }
+    {
+      name: invalid_sync_pattern
```

Comment:
I don't understand this test. The 'usbdev' is required to ignore any transmission that does not commence with a valid SYNC pattern. There is no interrupt that reports this condition.

It may, however, be valuable to 'fuzz' the device by sending it randomized transmission that are known to contain valid SYNC patterns, to ensure that the device does not respond inappropriately or enter a bad internal state. This would test its resilience to connection issues, electrical noise and invalid transmissions.

The device should still be capable of responding to valid packet data after any such invalid transmissions.